core/ecschema-editing/src/Editing/Properties.ts

```diff
@@ -0,0 +1,271 @@
+    for (const derivedClass of derivedClasses) {
+      if (await derivedClass.getProperty(newPropertyName))
+        throw new Error(`An ECProperty with the name ${newPropertyName} already exists in the class ${derivedClass.fullName}.`);
+
+      const propertyOverride = await derivedClass.getProperty(existingPropertyName) as MutableProperty;
+      // If found the property is overridden in the derived class.
+      if (propertyOverride)
+        derivedProperties.push(propertyOverride);
+    }
+
+    // Re-name the overridden property in all derived classes
+    derivedProperties.forEach((prop: MutableProperty) => {
+      prop.setName(newName);
+    });
+
+    existingProperty.setName(newName);
+  }
+
+  /**
+   * Sets the property description.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param description The new description to set.
+   */
+  public async setDescription(classKey: SchemaItemKey, propertyName: string, description: string) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setDescription(description);
+  }
+
+  /**
+   * Sets the property label.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param label The new label to set.
+   */
+  public async setLabel(classKey: SchemaItemKey, propertyName: string, label: string) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setLabel(label);
+  }
+
+  /**
+   * Sets the property isReadOnly attribute.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param isReadOnly The new isReadOnly value.
+   */
+  public async setIsReadOnly(classKey: SchemaItemKey, propertyName: string, isReadOnly: boolean) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setIsReadOnly(isReadOnly);
+  }
+
+  /**
+   * Sets the property priority.
+   * @param classKey The SchemaItemKey of the class.
+   * @param propertyName The name of the property.
+   * @param priority The new priority.
+   */
+  public async setPriority(classKey: SchemaItemKey, propertyName: string, priority: number) {
+    const property = await this.getProperty<MutableProperty>(classKey, propertyName);
+    property.setPriority(priority);
+  }
+
+  private async findDerivedClasses(mutableClass: MutableClass): Promise<Array<MutableClass>>{
+    const derivedClasses: Array<MutableClass> = [];
+    const schemaItems = this._schemaEditor.schemaContext.getSchemaItems();
+    let { value, done } = schemaItems.next();
+    while (!done) {
+      if (await value.is(mutableClass)) {
+        if (!mutableClass.key.matches(value.key)) {
+          derivedClasses.push(value);
+        }
+      }
+      ({ value, done } = schemaItems.next());
+    }
```

Comment:
Wouldn't a async for-loop work too here?
Also value.is only exists for classes, but getSchemaItems could return any kind of schema items, so we might to add a typecheck here too.
```ts
for await (const schemaItem of this._schemaEditor.schemaContext.getSchemaItems()) {
  if(ECClass.isECClass(schemaItem) && await schemaItem.is(mutableClass)) {
    // ...
  }
}
```

Reply:
yes, much cleaner. thanks.